internal/core/window.rs

```diff
@@ -205,6 +205,60 @@ struct PopupWindow {
     component: ComponentRc,
 }
 
+#[derive(Default)]
+struct ClickState {
+    click_count_time_stamp: Cell<Option<crate::animations::Instant>>,
+    click_count: Cell<u8>,
+}
+
+impl ClickState {
+    fn reset(&self) {
+        self.click_count.set(0);
+        self.click_count_time_stamp.set(Some(crate::animations::Instant::now()));
+    }
+    fn check_repeat(&self, mut mouse_event: MouseEvent) -> MouseEvent {
+        match mouse_event {
+            MouseEvent::Pressed { position, button, .. } => {
+                if let Some(click_count_time_stamp) = self.click_count_time_stamp.get() {
+                    if crate::animations::Instant::now() - click_count_time_stamp
```

Comment:
You can cache the value of crate::animations::Instant::now() so you don't call it again 6 lines down.

Reply:
done